### PR TITLE
ROX-23386: Add PlatformCVECountByType and PlatformCVECountByFixability; rename ClusterCountByPlatformType to match UI implementation

### PR DESCRIPTION
**Pull request by charmik-redhat**
## Description

- Rename ClusterCountByPlatformType resolver to ClusterCountByType so that it matches the query called by UI implementation
- Add PlatformCVECountByType resolver
- Add PlatformCVECountByFixability resolver

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

### Here I tell how I validated my change

- Unit tests
- Manual tests by calling the newly added queries

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: central/graphql/resolvers/clusters.go [hunk 1]
```diff
@@ -953,3 +956,41 @@ func (resolver *clusterResolver) UnusedVarSink(_ context.Context, _ RawQuery) *i
 func (resolver *orchestratorMetadataResolver) OpenShiftVersion() (string, error) {
 	return resolver.data.GetOpenshiftVersion(), nil
 }
+
+// PlatformCVECountByType returns resolver to get number of platform CVEs by CVE type
```
Comment (SimonBaeumer) [inline central/graphql/resolvers/clusters.go:960]: For my education: What does a resolver do?

Comment (charmik-redhat) [reply central/graphql/resolvers/clusters.go:960]: If there is a graphQL query like

```
cluster(id) {
    PlatformCVEsCountByType {
        Kubernetes
        Openshift
        Istio
    }
}
```

Here Kubernetes, Openshift and Istio are of `integer` type. But `cluster(id)` returns a complex type on which you can query other things like `PlatformCVEsCountByType`. Similarly `PlatformCVEsCountByType` is also a complex type on which you can query other things until you reach a scalar type like int or string. In graphQL implementations in our code base, we call these complex types resolvers. But they are just interfaces really.

File: central/views/platformcve/types.go [hunk 2]
```diff
@@ -38,3 +40,16 @@ type ClusterCountByPlatformType interface {
...
+
+// CVECountByType provides the number of platform CVEs of each type
+type CVECountByType interface {
+	GetKubernetesCVECount() int
+	GetOpenshiftCVECount() int
+	GetIstioCVECount() int
+}
+
+// CVECountByFixability provides the number of fixable and total platform CVEs
+type CVECountByFixability interface {
```
[the remark this answers is not in the document]
Comment (c-du) [reply central/views/platformcve/types.go:52]: Other than this, looks good to me.

[the remark this answers is not in the document]
Comment (charmik-redhat) [reply central/views/platformcve/types.go]: Good point. Changed it

**Review by c-du — APPROVED** [review]

**Pull request merged**